### ISSUE TITLE
content for accessioning is written with permissions that don't play well with forthcoming pres robots change

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/was_robot_suite#583]
**Describe the bug**

It seems (at first glance) that the was_robot_suite is writing warc files to `/dor/export/` with permissions that don't play well with the switch from tarpipe to `cp`ing from a shared mount for the pres robots `transfer-object` step.  see https://github.com/sul-dlss/preservation_robots/pull/461/ and https://github.com/sul-dlss/preservation_robots/pull/445

the error message seems very similar to a permissions issue we just debugged for Goobi for this pres robots change, see https://github.com/sul-dlss/preservation_robots/issues/458

**User Impact**

This prevents the switch to the more standard and more performant approach of copying from a shared mount, instead of tarpiping content over SSH.

**To Reproduce**
1. deploy the pres robots branch from https://github.com/sul-dlss/preservation_robots/pull/461/ to stage
2. run `web_archive_accessioning_spec.rb` from infra integration test suite against stage
3. see an error like `Error: transfer-object : Error transferring bag (via /dor/export/) for druid:kr060rx0708: Transfering bag for druid:kr060rx0708 to preservation failed. STDOUT = cp: cannot open '/dor/export/kr060rx0708/data/content/167545698620230203-8112-gx1jch.warc' for reading: Permission denied` on the test object's argo page, and a failure of the infra integration test

**Expected behavior**

Step 3 of steps to reproduce results in a successful test run and a successfully accessioned object, instead of an error

**Screenshots**

**Additional context**

i don't know this part of SDR super well, so apologies if my first guess as to the culprit codebase was wrong -- could also be was-registrar-app? 🤷  but this seemed to be the repo that does the writing to the DOR assembly space, see https://github.com/sul-dlss/was_robot_suite/blob/main/config/settings.yml#L15

**Comment by lwrubel** [comment]
The crawls get copied from `/was_unaccessioned_data/` to `/dor/workspace` in the wasCrawlPreassemblyWF: https://github.com/sul-dlss/was_robot_suite/blob/84255a29c1bb2595b8189d31eb5e858184c3a9e2/lib/robots/dor_repo/was_crawl_preassembly/build_was_crawl_druid_tree.rb#L25

**Comment by andrewjbtw** [comment]
Chalk another one up for the integration tests.

**Comment by jmartin-sul** [comment]
i'm wondering if there's a similar `umask` fix to what was done for the similar goobi issue in https://github.com/sul-dlss/preservation_robots/issues/458

yesterday, i glanced quickly at the perms for the object from my failed test:
```
was@was-robots1-stage:~$ ls -lah /dor/export/kr060rx0708/data/content/167545698620230203-8112-gx1jch.warc 
lrwxrwxrwx 1 was was 85 Feb  3 12:44 /dor/export/kr060rx0708/data/content/167545698620230203-8112-gx1jch.warc -> /dor/workspace/kr/060/rx/0708/kr060rx0708/content/167545698620230203-8112-gx1jch.warc
was@was-robots1-stage:~$ 
was@was-robots1-stage:~$ ls -lah /dor/workspace/kr/060/rx/0708/kr060rx0708/content/167545698620230203-8112-gx1jch.warc
-rw------- 1 was was 1.3M Feb  3 12:43 /dor/workspace/kr/060/rx/0708/kr060rx0708/content/167545698620230203-8112-gx1jch.warc
was@was-robots1-stage:~$ 
```

the symlink in `/dor/export` seems fine, but the perms on the file it links to seem too restrictive (should be group and world readable?).

if i look at a random warc from feb 1, i see a similar issue under the dir path @lwrubel mentions:
```
was@was-robots1-stage:~$ ls -lah /was_unaccessioned_data/jobs/1675271829
total 1.3M
drwxr-xr-x  2 was was 4.0K Feb  1 09:17 .
drwxr-xr-x 64 was was 8.0K Feb  7 11:54 ..
-rw-------  1 was was 1.3M Feb  1 09:17 167527182920230201-3251811-1w7u0n.warc
was@was-robots1-stage:~$ 
```

i'm not actively working on debugging this, but i did add it to the new column of the prod priorities board.

i think we can consider this fixed when the `web_archive_accessioning_spec` from infra integration tests passes with the branch from https://github.com/sul-dlss/preservation_robots/pull/461 deployed to the environment being tested.

**Comment by jmartin-sul** [comment]
i'll also observe that the containing dir in `/was_unaccessioned_data/jobs` is group and world readable:
```
drwxr-xr-x  2 was was 4.0K Feb  1 09:17 1675271829
```

**Comment by mjgiarlo** [comment]
It looks like the `was` user has a umask of `0002`, which should not cause 0600 perms.

**Comment by mjgiarlo** [comment]
@andrewjbtw It looks like you moved this from Ready back to New Issues. Is this no longer a problem we're seeing?

**Comment by andrewjbtw** [comment]
No, it's more that it's not pressing. This ticket is blocking moving pres-robots from using tar over ssh to using a shared mount and will still need to be done eventually. It doesn't seem like a good idea to change how pres-robots works while Infra is not on an SDR-related workcycle so I thought it would be best to take it out of ready. 

We don't have a backlog column on the priorities board so I wasn't sure where to put it.

**Comment by mjgiarlo** [comment]
I believe these files are initially created by wasapi_downloader from within the WRA fetch job. AFAICT, the downloader doesn't have any settings around initial perms. And the `was` user, which both the WRA web app and sidekiq process run under, has a umask of `0002` (same as on was_robot_suite VMs). so I'm not sure where the `0600` perms are coming from.

**Comment by mjgiarlo** [comment]
It would be pretty easy to throw a patch at WRA to manually `chmod` the files created by the downloader so that they're `0644` instead of `0600`. It would be even better to understand where the `0600` perms are coming from...

**Comment by mjgiarlo** [comment]
I think my analysis above was correct about how these files are created, but what I missed is that this error may not be "real:" it could be entirely caused by integration test setup, which I've now patched. This is ripe for testing in a deployed environment now.

**Comment by mjgiarlo** [comment]
@andrewjbtw tested in stage with https://github.com/sul-dlss/preservation_robots/pull/461 deployed and reports success.
